staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -888,9 +894,40 @@ type cacheWatcher struct {
 	stopped   bool
 	forget    func()
 	versioner storage.Versioner
+	timer     *time.Timer
+	timeout   time.Duration
+	created   time.Time
+}
+
+
+// var timerPool sync.Pool
+
+func newTimer(d time.Duration) *time.Timer {
+	return time.NewTimer(d)
+	/*
+	t, ok := timerPool.Get().(*time.Timer)
+	if ok {
+		t.Reset(d)
+	} else {
+		t = time.NewTimer(d)
+	}
+	return t
+	*/
 }
 
-func newCacheWatcher(chanSize int, filter filterWithAttrsFunc, forget func(), versioner storage.Versioner) *cacheWatcher {
+func freeTimer(timer *time.Timer) {
+	if !timer.Stop() {
+		// Consume triggered (but not yet received) timer event
+		// so that future reuse does not get a spurious timeout.
+		select {
+		case <-timer.C:
+		default:
```

Comment:
I'm not sure if this is the reason of the flakes, but this one is incorrect - you can't have default here.
If you want this to not generate spurious timeout for future use, you have to unconditionally wait for <-timer.C

Reply:
That is the case.  Add a default here because the timer.C may have been received by the process goroutine.

Reply:
This timer is created once, reuse is not necessary as in add(). WDYT?